sql/core/src/main/scala/org/apache/spark/sql/jdbc/OracleDialect.scala

```diff
@@ -68,5 +69,13 @@ private case object OracleDialect extends JdbcDialect {
     case _ => None
   }
 
+  override def beforeFetch(connection: Connection, properties: Map[String, String]): Unit = {
+    // Set general date and timestamp format before query.
+    val stmt = connection.createStatement()
+    stmt.execute("alter session set NLS_DATE_FORMAT = 'YYYY-MM-DD'")
+    stmt.execute("alter session set NLS_TIMESTAMP_FORMAT = 'YYYY-MM-DD HH24:MI:SS.FF'")
```

Comment:
If the ${option.table} is a subquery, then the NLS* setttings will work on the Oracle table in this subquery which containing the timestamp/date-related predicates. 

Specifically, these options only affect the Oralce JDBC sessions.